### PR TITLE
allow Authority as relationship name

### PR DESCRIPTION
**Pull request by xetys**
required for: https://github.com/jhipster/generator-jhipster/issues/11062

Please make sure the below checklist is followed for Pull Requests.
  - [ ] Checks are green
  - [ ] Tests are added where necessary
  - [ ] Documentation is added/updated where necessary
  - [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-core/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Can you add some unit tests plz ?

File: lib/validators/relationship_validator.js [hunk 1]
```diff
@@ -145,10 +146,11 @@ function checkManyToOneRelationship(jdlRelationship, skippedUserManagementOption
 }
 
 function checkManyToManyRelationship(jdlRelationship) {
-  const destinationEntityIsTheUser = isEntityTheUser(jdlRelationship.to);
+  const destinationEntityIsTheUser = isUserManagementEntity(jdlRelationship.to);
   if (jdlRelationship.injectedFieldInFrom && !jdlRelationship.injectedFieldInTo && destinationEntityIsTheUser) {
     // This is a valid case: even though bidirectionality is required for MtM relationships, having the destination
     // entity being the User is possible.
+    // throw new Error(`test ${jdlRelationship.to}`);
```
Comment (MathieuAA) [inline lib/validators/relationship_validator.js]: can you remove this line please?

**Comment by MathieuAA** [comment]
Where are we on this?

**Comment by pascalgrimaud** [comment]
@MathieuAA : still missing unit tests I think

**Comment by xetys** [comment]
I'll will proceed this topic this week. Just back from winter vacation

**Comment by MathieuAA** [comment]
@xetys nice! thanks

**Comment by xetys** [comment]
sry, took some time, hope the very basic tests inspired by the user relation tests are fine 

If something is missing, feel free to ask for changes

**Comment by MathieuAA** [comment]
Thanks @xetys :)

**Pull request merged**